### ISSUE TITLE
SPIR-V shader entry point not found when targeting vulkan 1.1 or above

### ISSUE THREAD

**Issue by tim-rex** [KhronosGroup/MoltenVK#2002]
I have SPIR-V shaders built using glslc

By default, glslc will target vulkan 1.0 and all seems well.
When I attempt to produce SPIR-V intermediates targeting Vulkan 1.1 or above, I hit validation errors in MoltenVK regarding a missing entry point.

I couldn't find any documentation to indicate that Molten VK only supports SPIR-V targeting Vulkan 1.0 (perhaps I didn't look hard enough). It's also possible this is an issue with glslc, though I can confirm the entry points are visible and reported by spirv-cross.

Example usage:
`/usr/local/bin/glslc --target-env=vulkan1.1 hello_triangle.vert -o hello_triangle.vert.spv`

Validation error:
```
validation layer: (Validation) Validation Error: [ VUID-VkPipelineShaderStageCreateInfo-pName-00707 ] | MessageID = 0xb38b9761 | vkCreateGraphicsPipelines(): pCreateInfos[0] No entrypoint found named `main` for stage VK_SHADER_STAGE_VERTEX_BIT. The Vulkan spec states: pName must be the name of an OpEntryPoint in module with an execution model that matches stage (https://vulkan.lunarg.com/doc/view/1.3.261.0/mac/1.3-extensions/vkspec.html#VUID-VkPipelineShaderStageCreateInfo-pName-00707)
[mvk-error] VK_ERROR_INITIALIZATION_FAILED: Failed to get vertex outputs: Entry point does not exist.
validation layer: (Validation) VK_ERROR_INITIALIZATION_FAILED: Failed to get vertex outputs: Entry point does not exist.
```

Vulkan SDK: 1.3.261
glslc version reports: 
```
/usr/local/bin/glslc --version
shaderc v2023.6 v2023.6
spirv-tools v2023.4 v2022.4-296-ge553b884
glslang 11.1.0-763-g76b52ebf

Target: SPIR-V 1.0
```

**Comment by cdavis5e** [comment]
Are you sure this is a bug in MoltenVK? The Vulkan validation layer is also complaining, which usually means you're holding it wrong. ;) Does the SPIR-V module work with some other Vulkan driver? (You may need to try a different platform to test that.)

Are you sure that `shaderc` named the entry point in the SPIR-V module `main`? What does the SPIR-V look like? (You can use the `spirv-dis` tool from the SPIRV-Tools distribution to inspect a SPIR-V module.)

**Comment by tim-rex** [comment]
Thanks @cdavis5e 
Turns out I was doing something stupid with my resource loader (hooray for unsafe string pointers)..  resulting in the pipeline being built with both the vertex and shader stages being supplied with a single vertex spirv intermediate.

I found some other wierdness with my build that might account for why it _seemed_ to work when targeting vulkan1.0 only.

Apologies for the wasted effort.. definitely user error on my part.
